sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -371,6 +371,10 @@ public void SetStartTime(DateTime startTime)
             public void MarkFailed(Exception exception)
             {
                 _diagnosticSource?.Write(_activityName + ".Exception", exception);
+                if (ActivityExtensions.SupportsActivitySource())
+                {
+                    _currentActivity?.SetErrorStatus(exception.ToString());
```

Comment:
does it also affect HTTP requests which return 4xx, 5xx? I.e. when we retry those, we should see

- outer logical activity  - error (after all tries)
  - HTTP try 1  - (DNS exception) - error 
  - HTTP try 2  - (503)  - error 
  - HTTP try 3  - (429)  - error 
  - .etc